src/neo-vm/Types/Struct.cs

```diff
@@ -31,9 +32,11 @@ public Struct(ReferenceCounter? referenceCounter, IEnumerable<StackItem>? fields
         /// <summary>
         /// Create a new structure with the same content as this structure. All nested structures will be copied by value.
         /// </summary>
+        /// <param name="limits">Execution engine limits</param>
         /// <returns>The copied structure.</returns>
-        public Struct Clone()
+        public Struct Clone(ExecutionEngineLimits limits)
         {
+            var count = limits.MaxInvocationStackSize - 1;
```

Comment:
BTW, that's for invocation stack, I'm not sure using it for anything item-related is appropriate.

Reply:
```suggestion
            var count = limits.MaxStackSize - 1;
```